pkg/allocator/allocator.go

```diff
@@ -480,13 +488,19 @@ func (a *Allocator) lockedAllocate(ctx context.Context, key AllocatorKey) (idpoo
 		if value != 0 {
 			// re-create master key
 			if err := a.backend.UpdateKeyIfLocked(ctx, value, key, true, lock); err != nil {
-				return 0, false, fmt.Errorf("unable to re-create missing master key '%s': %s while allocating ID: %s", key, value, err)
+				return 0, false, false, fmt.Errorf("unable to re-create missing master key '%s': %s while allocating ID: %s", key, value, err)
 			}
 		}
 	} else {
-		_, err := a.localKeys.allocate(k, key, value)
+		_, firstUse, err = a.localKeys.allocate(k, key, value)
 		if err != nil {
-			return 0, false, fmt.Errorf("unable to reserve local key '%s': %s", k, err)
+			return 0, false, false, fmt.Errorf("unable to reserve local key '%s': %s", k, err)
+		}
+
+		if firstUse {
+			log.WithField(fieldKey, k).Info("Reserved new local key")
+		} else {
+			log.WithField(fieldKey, k).Info("Reusing existing local key")
```

Comment:
Are these info messages really necessary? They can flood the logs on startup and they seem more like debugging information for developers rather than informational messages for users.

Reply:
Hi @joestringer, turning them into debug level is OK. You're right, they are more like debugging information.

Reply:
I sent a PR for this, PTAL: https://github.com/cilium/cilium/pull/14804